### ISSUE TITLE
Sonoff Touch Testing

### ISSUE THREAD

**Issue by BrettSheleski** [BrettSheleski/SmartThingsPublic#12]
Please use this thread for communicating issues with the Sonoff Touch module.

**Comment by BornInThe50s** [comment]
Hi Brett, I've flashed a T1 2 Channel, UK variant with Tasmoto 5.12. 
Installed the latest Device Handlers and App from your GitHub  into the SmartThings IDE.
I added the T1 into SmartThings via the add device in the IDE, gave it the IP of the T1 and it installed, it then created the two child devices for the switches. They work as expected, however, it looks odd in the IDE and the App. This is because it creates the parent ie Kitchen Light Switch, then creates the two child devices, Kitchen Light Switch Power 1 and Kitchen Light Switch Power 2. The Power 1 and 2 control the touch switches fine, the parent shows as being 'ON' but doesn't control anything and can't be turned off.
Ideally the parent needs hiding, or, could it be utilised so that it powers off/on both child devices like a group?
Great to see someone is getting Tasmota integrated into SmartThings like this, big thumbs up from me.

**Comment by BrettSheleski** [comment]
You're correct in that the parent device is not intended to be used directly by the user for controlling the channel(s) of the Sonoff device, that is what the child devices would be used for (as you have observed).

I personally don't think that the parent device should be hidden in the SmartThings App as it does provide some information to the user.  At this points I haven't cared a great deal of what it displays in the App.  Currently it just lists off some potentially useful information (IP address, MAC address, etc.) that should be available for all Tasmota devices.  It showing 'ON' in the app is not ideal and probably misleading.  I'll see about remove that part at least.

It would be nice for the SmartThings app to display all child devices.  I don't know if that is possible given the limitations of the SmartThings API.

Glad to hear you got it working on the T1 (2 Channel), I'll update the table in the readme.md with you as the tester.

Feel free to submit pull requests for any changes you would like to see.

**Comment by velkovb** [comment]
I just flashed an EU T1 2 channel and it seems to be working properly but one thing I noticed is that it is reported its status back to SmartThings too slowly. Is there a way this can be fixed?

My case is that I want to link the Sonoff to another switch so they can operate like 2-way switch but the Sonoff has like several seconds delay.

**Comment by BrettSheleski** [comment]
There is a roundabout way of getting the status to report back to SmartThings using a SmartApp I developed along with using Node-Red.  It's not pretty, but it works.

I'd like to implement SmartThings integration directly into the Tasmota firmware, but that's not going to happen anytime soon.

**Comment by ak47uk** [comment]
I have got this working on my T1 1-gang UK switch and 3 of my T1 3-gang UK switches. I ran a Tasmota FW upgrade on all switches before attempting this so they are on 6.6.0 (current).
One of my 3-gang switches would not spawn the child devices, I repeated the setup a few times and eventually, it worked once I used a different name. May have been a coincidence, I didn't use any special characters etc.
